xbmc/network/DNSNameCache.h

```diff
@@ -51,13 +51,10 @@ class CDNSNameCache final
    *
    * \return true if the IP is cached or the DNS query was successful
    */
-  static bool Lookup(const std::string& strHostName, std::string& strIpAddress);
+  bool Lookup(const std::string& strHostName, std::string& strIpAddress);
```

Comment:
Could this member function be `const`now?

Reply:
Only if I make `m_hostToIp` `mutable`. And even then it's questionable as this function changes the state of the object:
```
cache.GetCached("example.com", ...); // fails
cache.Lookup("example.com", ...); // succeeds
cache.GetCached("example.com", ...); // now succeeds despite only calling `const` functions
```

Reply:
Correct, but this isn't logical const. The contract for const is that if I only call const functions, then they should behave identical no matter how often or in which order they are called. This isn't the case here. But for `GetCached` it would be OK.

(At least that's my understanding :slightly_smiling_face:)

Reply:
https://stackoverflow.com/questions/3830367/difference-between-logical-and-physical-const-ness

There is exactly the cache lookup case mentioned for good use of a mutable class member: "A mutable class member can be modified even if the class object itself (or the reference or pointer used to access it) is const. Examples of good uses of this are a mutex that must be locked during a read operation, and a cache to store the result of an expensive read operation. In both cases, the operation itself should be a const function (since it doesn't affect the visible state of the object)..."

But no hard feelings, as this can for sure be seen from different perspectives.